week-2/Homework/mandatory/2-function-creation.js

```diff
@@ -26,7 +45,13 @@ The function must:
 */
 
 function remove(arr, index) {
-  return; // complete this statement
+  let newArray = [];
+  for (i in arr){
+    if (!(arr[i] == arr[index])){
+       newArray.push(arr[i]);
+    }
+  };
+  return newArray;
```

Comment:
Are you sure that this function would pass in any case? What if the same element appears in the array more than one time?

Reply:
Hi, if you notice this function takes 2 parameters, the second one is a number (index value ) not a value of the element. I believe indexes are uniqe in an array even if the elements of the array are not.

Reply:
I'll take a 2nd look at, might have missed something.

Reply:
You're right, the index is unique and you get an index number as the 2nd parameter.
So let's revise the criteria once more especially this one:
 * `remove the item at the specified index`

Do you think the condition `if (!(arr[i] == arr[index]))`, you have, is checking the position of the element in the array?

What do you think, what should be the outcome of the following example?
`remove(["a", "a", "a", "a", "a", "a", "a"], 6)`

Reply:
Hi, it is returning an empty array !!! after I ran this test I realised that this condition is comparing values not indexes. I'll have to research a bit further on this one

Reply:
I figured it out, whenever I write `arr[i]` I'll be accessing the value of the item in that index - which I don't want for the purposes of this function- the correct condition should be `if (!(i == index))` this way I'm only dealing with indexes.
This took me a while to understand. Thanks again Zoltan